.github/workflows/pr-checks.yaml

```diff
@@ -16,6 +16,49 @@ env:
   TEST_RESULTS: "out/acceptance-tests"
 
 jobs:
+  lint:
+    name: Code Quality
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "^1.19"
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          architecture: "x64"
+
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Lint Go
+        if: success() || failure()
+        run: make lint-go
+
+      - name: Lint YAML files
+        if: success() || failure()
+        run: make lint-yaml
+
+      - name: Lint Python
+        if: success() || failure()
+        run: make lint-python
+
+      - name: Lint Gherkin (feature) files
+        if: success() || failure()
+        run: make lint-feature-files
+
+      - name: Lint conflicts in source files
+        if: success() || failure()
+        run: make lint-conflicts
+
+      - name: Lint shell scripts
+        if: success() || failure()
+        run: make lint-shell
```

Comment:
Why keep these as separate steps?  `make lint` will do roughly the same things, so what's the difference?

Reply:
@sadlerap, you are correct, the `make lint` runs it all. However, it runs it all in one big chunk and if any of the linters fails, it is tedious to go through the large log to investigate. By splitting it into individual steps, it is immediately clear which one of the linters failed and much easier to investigate or debug just the relevant log.

Reply:
Additionally, by using the "split" approach all the linters are executed even if some of them failed, while in case of using `make lint`, the execution would fail fast as soon as any linter failed, actually skipping the remaining ones. So if multiple linters are set to fail, you would always see just the first one that faied, missing the others.

`make -k lint` would fix that but you would still have to go through a large overall log to find the actual problem.